frontend/src/component/common/Search/Search.tsx

```diff
@@ -148,7 +149,8 @@ export const Search = ({
                             <Tooltip title="Clear search query" arrow>
                                 <IconButton
                                     size="small"
-                                    onClick={() => {
+                                    onClick={e => {
+                                        e.stopPropagation(); // prevent outside click from the lazily added element
```

Comment:
search container ref couldn't see the icon button that is added in a conditional render. As a result the IconButton is not considered to be part of the searchContainerRef. It means that it was considered as a click outside of the searchContainerRef which closes the search. We don't want that so we stop event propagation.